Mods/DiscordRichPresence/Src/DiscordRpc.cpp

```diff
@@ -0,0 +1,203 @@
+		{ "salty", "Hantu Port" },
+		{ "caged", "Siberia" },
+		// Prologue
+		{ "thefacility", "ICA Facility" },
+		// HITMAN
+		{ "paris", "Paris" },
+		{ "coastaltown", "Sapienza" },
+		{ "marrakesh", "Marrakesh" },
+		{ "bangkok", "Bangkok" },
+		{ "colorado", "Colorado" },
+		{ "hokkaido", "Hokkaido" },
+		// HITMAN 2
+		{ "sheep", "Hawke's Bay" },
+		{ "miami", "Miami" },
+		{ "colombia", "Santa Fortuna" },
+		{ "mumbai", "Mumbai" },
+		{ "skunk", "Whittleton Creek" },
+		{ "theark", "Isle of Sgàil" },
+		{ "greedy", "New York" },
+		{ "opulent", "Haven Island" },
+		// HITMAN 3
+		{ "golden", "Dubai" },
+		{ "ancestral", "Dartmoor" },
+		{ "edgy", "Berlin" },
+		{ "wet", "Chongqing" },
+		{ "elegant", "Mendoza" },
+		{ "trapped", "Carpathian Mountains" }
+	};
+	Logger::Info("Finished populating scene map");
+}
+
+void DiscordRpc::PopulateGameModes()
+{
+	m_gameModes = {
+		{ "sniper", "Sniper Assassin" },
+		{ "usercreated", "Contracts Mode" },
+		{ "creation", "Contracts Mode" },
+		{ "featured", "Featured Contract" },
+		{ "mission", "Mission" },
+		{ "flashback", "Mission" },
+		{ "tutorial", "Mission" },
+		{ "campaign", "Mission" },
+		{ "escalation", "Escalation" },
+		{ "elusive", "Elusive Target" }
+	};
+	Logger::Info("Finished populating game modes");
+}
+
+void DiscordRpc::PopulateCodenameHints()
+{
+	m_codenameHints = {
+		{ "GarterSnake", "A Bitter Pill" },
+		{ "Spider", "A Gilded Cage" },
+		{ "Python", "A House Built On Sand" },
+		{ "Cottonmouth", "A Silver Tongue" },
+		{ "Skunk", "Another Life" },
+		{ "Edgy Fox 4", "Apex Predator" },
+		{ "Polarbear Arrival FSP", "Arrival" },
+		{ "Magpie", "The Ark Society" },
+		{ "Mongoose", "Chasing A Ghost" },
+		{ "Tiger", "Club 27" },
+		{ "Bulldog", "Death In The Family" },
+		{ "Anaconda", "Embrace Of The Serpent" },
+		{ "Rat", "End Of An Era" },
+		{ "Flamingo", "The Finish Line" },
+		{ "Bull", "Freedom Fighters" },
+		{ "Polarbear Module 002_B", "Freeform Training" },
+		{ "Greedy Raccoon 16", "Golden Handshake" },
+		{ "Polarbear Module 002 FSP", "Guided Training" },
+		{ "Mamushi", "Hokkaido Snow Festival" },
+		{ "Paris Noel", "Holiday Hoarders" },
+		{ "KingCobra", "Illusions Of Grandeur" },
+		{ "Mamba", "Landslide" },
+		{ "Sheep 9", "Nightcall" },
+		{ "Golden Gecko", "On Top Of The World" },
+		{ "Flu", "Patient Zero" },
+		{ "Snow Crane", "Situs Inversus" },
+		{ "Ebola", "The Author" },
+		{ "Llama", "The Farewell" },
+		{ "Polarbear Graduation", "The Final Test" },
+		{ "Copperhead", "The Icon" },
+		{ "Stingray", "The Last Resort" },
+		{ "Peacock", "The Showstopper" },
+		{ "Zika", "The Source" },
+		{ "Hippo", "Three-Headed Serpent" },
+		{ "Wolverine", "Untouchable" },
+		{ "Rabies", "The Vector" },
+		{ "Octopus", "World Of Tomorrow" },
+		// Sniper Assassin
+		{ "SC_Hawk", "The Last Yardbird" },
+		{ "SC_Seagull", "The Pen And The Sword" },
+		{ "SC_Falcon", "Crime And Punishment" }
+	};
+}
+
+std::string DiscordRpc::LowercaseString(std::string in)
```

Comment:
String parameters should be passed as `const std::string&` or `const ZString&` (a const reference) to prevent making unnecessary copies.